test/fixtures/define-properties/case-2/result.js

```diff
@@ -0,0 +1,63 @@
+const assert = require('assert');
+
+
+const $define = function (object, map) {
+  let propKeys = Object.keys(map);
+  propKeys = propKeys.concat(Object.getOwnPropertySymbols(map));
+
+  for (var i = 0; i < propKeys.length; i += 1) {
+    const propKey = propKeys[i];
+    const value = map[propKey];
+
+    if (propKey in object) {
+      continue;
+    }
+
+    Object.defineProperty(object, propKey, {
+      value,
+      configurable: true,
+      enumerable: false,
+      writable: false,
+    })
+  }
+
+  return object;
+};
+
+const object1 = { a: 1, b: 2 };
+
+const res1 = $define(object1, {
```

Comment:
Many of these test cases came from the `define-properties` [readme](https://www.npmjs.com/package/define-properties?activeTab=readme) and the [repo itself](https://github.com/ljharb/define-properties/blob/main/test/index.js)


One thing I need to investigate further is the correct descriptor configuration. 

In the README, the created descriptor is asserted as being:

```
{
  configurable: true,
  enumerable: false,
  value,
  writable: false
}
```

Whereas in the repo's test case, it's asserted as being:

```
{
  configurable: true,
  enumerable: false,
  value,
  writable: true
};
```

Reply:
Judging from the [`define-data-property`](https://github.com/ljharb/define-data-property/blob/main/index.js) repo, a descriptor is non-writable by default, and this can be overridden using the 5th argument of the function.

From the looks of the [`define-properties`](https://github.com/ljharb/define-properties/blob/main/index.js#L27) repo a 5th argument is passed through, so it should always be `false`. I don't know why the [`define-properties`](https://github.com/ljharb/define-properties/blob/main/test/index.js#L22) test suggests otherwise.

Reply:
This has been resolved: https://github.com/es-tooling/module-replacements-codemods/issues/66#issuecomment-2258854753